### DISCUSSION TITLE
user.suspended_at - GraphQL/REST API Permission Mismatch

### DISCUSSION THREAD

**Discussion by MarkusMit** [community/community#77087] [Enterprise]
### Select Topic Area

Bug

### Body

>  (GitHub Enterprise Server 3.8)

If I try to to access `User.suspendedAt` via the GitHub **GraphQL** API I get a "required scopes" error:

```graphQL
query FindUsers ($myQuery: String!) {
	search (query: $myQuery type: USER first: 10) {
		userCount
		nodes {
			... on User {
				login
				name
				email
				suspendedAt
			}
		}
	}
}
```

Error:
> "Your token has not been granted the required scopes to execute this query. The 'suspendedAt' field requires one of the following scopes: ['site_admin'], but your token has only been granted the: ['admin:org', 'repo', 'user:email'] scopes. Please modify your token's scopes at: https://github.developer.allianz.io/settings/tokens."

The error is the same regardless of I perform a '[search][4]' or a direct '[user][5]' query.

However, if I retrieve the user's data with a [**REST** call][3], using the same token as with the GraphQL call, I get that very information:
[![Snipet of REST response, showing "suspended_at" field with a timestamp][2]][2]

Also, although my user is no "site-admin" I see on tht users profile page on the GitHub **Web-UI** that the user has been suspended:
[![Screenshot snippet showing "This user is spuended."][1]][1]

As I said, I'm no `site_admin`, so I can't add that scope to my token. 
But I also don't understand why I would need that scope, when I can see the information on the UI.

*Note:* I would actually only care about the 'suspended' state. With my current use-case I do not need the timestamp of suspension.

  [1]: https://i.stack.imgur.com/ymIEK.png
  [2]: https://i.stack.imgur.com/GyB7S.png
  [3]: https://docs.github.com/en/enterprise-server@3.8/rest/users/users#get-a-user
  [4]: https://docs.github.com/en/enterprise-server@3.8/graphql/reference/queries#search
  [5]: https://docs.github.com/en/enterprise-server@3.8/graphql/reference/queries#user

**Comment by Naimulnobel** [comment]
The `suspendedAt` field in the GitHub GraphQL API requires the `site_admin` scope, as the error message indicates. This is a requirement set by GitHub and there's no way around it within the GraphQL API.

The REST API and the GraphQL API have different permission models, which is why you're able to see the `suspended_at` field in the REST API response but not in the GraphQL API response.

As for the GitHub UI, it's designed to show a lot of information that might not be directly accessible via the API, or might require different permissions in the API. The UI and the API have different considerations and constraints, so they don't always align perfectly in terms of what information they show.

If you only care about the 'suspended' state and not the timestamp of suspension, you might be able to use the `isSuspended` field in the GraphQL API, if it's available in your GitHub Enterprise Server version. This field also requires the `site_admin` scope, but it's worth checking in case it's different in your specific version.

If that doesn't work, your best bet would be to continue using the REST API to get this information, since it seems to work with your current token and permissions.

**Reply by MarkusMit** [reply]
Thanks for your elaborate answer, @Naimulnobel.
However, I don't see any reason in having so severe divergences in the permission models of two different APIs for the same system, when they can be used with the same credentials (=personal access token). 
What is the reasoning behind requiring `site_admin` scope for a field on GraphQL API, when that strict permission model can be easily circumvented by retrieving the very same information via the REST API with way 'lighter' scopes? It doesn't make any sense from my point of view.
IMHO this is something that should be reconsidered by GitHub.

Of course, for the time being I'm using the REST API, since there's no way for me to receive 'site admin' permissions on the GitHub Enterprise server in question. Just wanted to point out this IMHO unreasonable divergence.
